src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -29,36 +29,21 @@ internal sealed class NameSyntaxClassifier : AbstractNameSyntaxClassifier
         SegmentedList<ClassifiedSpan> result,
         CancellationToken cancellationToken)
     {
-        if (syntax is NameSyntax name)
-        {
+        if (syntax is SimpleNameSyntax name)
             ClassifyTypeSyntax(name, semanticModel, result, cancellationToken);
-        }
     }
 
     public override ImmutableArray<Type> SyntaxNodeTypes { get; } =
     [
-        typeof(AliasQualifiedNameSyntax),
-        typeof(GenericNameSyntax),
         typeof(IdentifierNameSyntax),
-        typeof(QualifiedNameSyntax),
-        typeof(SimpleNameSyntax),
+        typeof(GenericNameSyntax),
```

Comment:
no need to check these other 'composed name types'.  We only care about `X` and `X<...>` (and we only classifiy the 'X' part of each).  Every other name type composes into one of these two name types that we actually classify here.

this addresses a *lot* of callbacks for dotted names that serve no purpose.